File: solutionJ60.java
public class solutionJ60 {
    public double[] dicesProbability(int n) {
        int[][] dp = new int[n][6 * n];
        for (int i = 0; i < 6; i++) {
            dp[0][i] = 1;
        }
        for (int i = 1; i < n; i++) {
            // 第 i + 1 个骰子
            // 投递 i + 1 个骰子产生的点数和的范围是 [i + 1, 6 * (i + 1)]
            for (int j = i; j < 6 * (i + 1); j++) {
                for (int k = 1; k < 7; k++) {
                    if (j - k >= 0) {
                        dp[i][j] += dp[i - 1][j - k];
                    }
                }
            }
        }
        double sum = Math.pow(6, n);
        double[] res = new double[5 * n + 1];
        for (int i = 0; i < res.length; i++) {
            res[i] = dp[n - 1][n + i - 1] / sum;
        }
        return res;
    }
}

File: solutionJ60.go
package leetcode

import (
	"fmt"
	"math"
)

func dicesProbability(n int) []float64 {
	dp := make([][]int, n)
	for i := 0; i < n; i++ {
		dp[i] = make([]int, 6*n)
	}
	for i := 0; i < 6; i++ {
		dp[0][i] = 1
	}
	for i := 1; i < n; i++ {
		for j := i; j < 6*(i+1); j++ {
			for k := 1; k < 7; k++ {
				if j-k >= 0 {
					dp[i][j] += dp[i-1][j-k]
				}
			}
		}
	}
	var sum = math.Pow(float64(6), float64(n))
	fmt.Print(sum)
	res := make([]float64, 5*n+1)
	for i := 0; i < 5*n+1; i++ {
		res[i] = float64(dp[n-1][n+i-1]) / sum
	}
	return res
}
